test/property.test.js

```diff
@@ -51,36 +75,215 @@ describe('loopback:property generator', function() {
       expect(props).to.have.property('isPreferred');
       expect(props.isPreferred).to.eql({
         type: 'boolean',
-        required: true,
         default: true
       });
       done();
     });
   });
 
+  it('creates date type property from ISO string', function(done) {
+    var propertyGenerator = givenPropertyGenerator();
+    helpers.mockPrompt(propertyGenerator, {
+      model: 'Car',
+      name: 'year',
+      type: 'date',
+      defaultValue: '2015-11'
+    });
+
+    propertyGenerator.run(function() {
+      var definition = common.readJsonSync('common/models/car.json');
+      var props = definition.properties || {};
+      expect(props).to.have.property('year');
+      expect(props.year).to.eql({
+        type: 'date',
+        default: '2015-11-01T00:00:00.000Z'
+      });
+      done();
+    });
+  });
+
+  it('creates date type property from number', function(done) {
+    var propertyGenerator = givenPropertyGenerator();
+    helpers.mockPrompt(propertyGenerator, {
+      model: 'Car',
+      name: 'year',
+      type: 'date',
+      defaultValue: '1466087191000'
+    });
+
+    propertyGenerator.run(function() {
+      var definition = common.readJsonSync('common/models/car.json');
+      var props = definition.properties || {};
+      expect(props).to.have.property('year');
+      expect(props.year).to.eql({
+        type: 'date',
+        default: '2016-06-16T14:26:31.000Z'
```

Comment:
Because I read the implementation, I assume that `2016-06-16T14:26:31.000Z` is same as timestamp `1466087191000`. However, readers not familiar with the implementation won't be able to tell that.

Please make the code easier to understand, for example by
 - changing this line to `default: new Date(1466087191000).toISOString()`
 - or adding a comment explaining how is the expected value related to the provided timestamp string